### PR TITLE
Add support for topk metadata transferring for PD

### PR DESCRIPTION
**Pull request by ShangmingCai**
## Motivation
In `process_prebuilt_extend` of the decode side, `topk_p` and `topk_index` are fake, limiting the accept rate improvement of transfer hidden states in the EALGE model.
```python
        # Simulate the eagle run. We add mock data to hidden states for the
        # ease of implementation now meaning the first token will have acc rate
        # of 0.
        if not self.spec_algorithm.is_none():

            b = len(self.reqs)
            topk_p = torch.arange(
                b * server_args.speculative_eagle_topk,
                0,
                -1,
                device=self.device,
                dtype=torch.float32,
            )
            topk_p = topk_p.reshape(b, server_args.speculative_eagle_topk)
            topk_p /= b * server_args.speculative_eagle_topk
            topk_index = torch.arange(
                b * server_args.speculative_eagle_topk, device=self.device
            )
            topk_index = topk_index.reshape(b, server_args.speculative_eagle_topk)

            hidden_states_list = [req.hidden_states_tensor for req in self.reqs]
            hidden_states = torch.stack(hidden_states_list, dim=0).to(self.device)

            # local import to avoid circular import
            from sglang.srt.speculative.eagle_utils import EagleDraftInput

            spec_info = EagleDraftInput(
                topk_p=topk_p,
                topk_index=topk_index,
                hidden_states=hidden_states,
                verified_id=self.output_ids,
            )
            spec_info.prepare_for_extend(self)
            spec_info.capture_hidden_mode = CaptureHiddenMode.LAST
            self.spec_info = spec_info
```

## Modifications
Transfer `topk_p` and `topk_index` from the prefill side, and set the value in `process_prebuilt_extend` of the decode side.

## Tests

> #10616 transfer the topk_p and topk_index, expected to improve acceptance rate in first decode draft forward, here is an experiment to show the improvement. The launch command is same as https://github.com/sgl-project/sglang/issues/10059#issuecomment-3257390658 , in benchmark we use a lower  `sharegpt-output-len` to show the acceptance rate improvement in first decode draft forward.
> 
> #### Benchmark command
> ```
> python3 /mnt/data/huangziming/sgl_dev/sglang/python/sglang/bench_serving.py \
>   --backend sglang-oai-chat \
>   --dataset-name sharegpt \
>   --output-details \
>   --num-prompt 1000 \
>   --max-concurrency 64 \
>   --sharegpt-output-len 4 \
>   --warmup-requests 0 \
>   --port 8000 \
>   --model /models/DeepSeek-R1-0528 \
> ```
> #### Results
> We compared the `avg_accept_length` before #9976 ,  after #9976 and set hidden states to `spec_info.hidden_states`
> 
> | before #10616 | after #10616 | after #10616 + use spec_info.hidden_states |
> |-------|-------|-------|
> | 1.362 | 1.503 | 1.565 |
> 
> The result shows transferring  `topk_p` and `topk_index` and use `spec_info.hidden_states` can improve acceptance rate 

 _Originally posted by @ZeldaHuang in [#10059](https://github.com/sgl-project/sglang/issues/10059#issuecomment-3322433179)_

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/disaggregation/decode_schedule_batch_mixin.py [hunk 1]
```diff
@@ -131,19 +131,29 @@ def process_prebuilt_extend(
         if not self.spec_algorithm.is_none():
```
Comment (ZeldaHuang) [inline python/sglang/srt/disaggregation/decode_schedule_batch_mixin.py]: maybe we should use `if self.spec_algorithm.is_eagle():` here, for none eagle speculative algorithm we don't need these。

Comment (hnyls2002) [reply python/sglang/srt/disaggregation/decode_schedule_batch_mixin.py]: agree

Comment (ShangmingCai) [reply python/sglang/srt/disaggregation/decode_schedule_batch_mixin.py]: Fixed.

**Review by hnyls2002 — APPROVED** [review]
